lib/abstract_format.ml

```diff
@@ -596,6 +603,58 @@ and expr_of_sf sf : (expr_t, err_t) Result.t =
      let%bind rhs = sf_rhs |> expr_of_sf |> track ~loc:[%here] in
      ExprBinOp {line; op; lhs; rhs} |> return
 
+  (* a record creation : #user{name = "Taro", admin = true} *)
+  | Sf.Tuple (4, [Sf.Atom "record";
+                  Sf.Integer line;
+                  Sf.Atom name;
+                  Sf.List sf_record_fields]) ->
+     let field_of_sf sf =
+       begin match record_field_of_sf sf with
+       | Ok (RecordField {line; field_name; ty=None; default_expr=Some e}) ->
+          (line, field_name, e) |> return
+       | Ok _ ->
+          Err.create ~loc:[%here] (Err.Invalid_input ("cannot reach here", sf)) |> Result.fail
```

Comment:
Could you make error message more detailed? e.g., "record field must have field name and expression" 🙏

Reply:
I see 🙏